File: Java_Programs/singlelevel.java

class singlelevel
{
    public static void main(String args[])
    {
        System.out.println("Inside main method");

        Derived dobj = new Derived();

        dobj.fun();
        dobj.gun();

    }
}
class Base
{
    public int a,b;

    public Base()
    {
        System.out.println("Inside Base Constructor");
    }
    public void fun()
    {
        System.out.println("Inside Base fun");
    }
}

class Derived extends Base
{
    public int x,y;

    public Derived()
    {
        System.out.println("Inside Derived Constructor");
    }
    public void gun()
    {
        System.out.println("Inside Derived gun");
    }
}

File: C_Programs/singlelevel.cpp
#include<iostream>
using namespace std;


class Base
{
    public:
        int a,b;

        Base()
        {
            cout<<"\nInside Base constructor";
            a = 10;
            b=20;
        }
        ~Base()
        {
            cout<<"\nInside Base Destructor";
        }

        void fun()
        {
            cout<<"\nInside Base Function";
        }

};

class Derived:public Base
{
    public:
        int x,y;

        Derived()
        {
            cout<<"\nInside Derived constructor";
            x=30;
            y=40;
        }
        ~Derived()
        {
            cout<<"\nInside Derived Destructor";
        }

        
        void gun()
        {
            cout<<"\nInside Derived Function";
        }
};


int main()
{

    Derived dobj;

    //cout<<sizeof(bobj)<<"\n";
    //cout<<sizeof(dobj)<<"\n";

    cout<<"\nInside main function"<<"\n";

    cout<<dobj.a<<"\n";
    cout<<dobj.b<<"\n";
    cout<<dobj.x<<"\n";
    cout<<dobj.y<<"\n";

    dobj.fun();
    dobj.gun();


    return 0;
}